cql3/restrictions/statement_restrictions.cc

```diff
@@ -1003,5 +1010,652 @@ ::shared_ptr<single_column_restriction> single_column_restriction::LIKE::apply_t
+            throw exceptions::invalid_request_exception("subscripting non-map column");
+        }
+        const auto deserialized = cdef->type->deserialize(*data.other_columns[data.sel.index_of(*cdef)]);
+        const auto& data_map = value_cast<map_type_impl::native_type>(deserialized);
+        const auto key = col.sub->bind_and_get(data.options);
+        auto&& key_type = col_type->name_comparator();
+        const auto found = with_linearized(*key, [&] (bytes_view key_bv) {
+            using entry = std::pair<data_value, data_value>;
+            return std::find_if(data_map.cbegin(), data_map.cend(), [&] (const entry& element) {
+                return key_type->compare(element.first.serialize_nonnull(), key_bv) == 0;
+            });
+        });
+        return found == data_map.cend() ? bytes_opt() : bytes_opt(found->second.serialize_nonnull());
+    } else {
+        switch (cdef->kind) {
+        case column_kind::partition_key:
+            return data.partition_key[cdef->id];
+        case column_kind::clustering_key:
+            return data.clustering_key[cdef->id];
+        case column_kind::static_column:
+        case column_kind::regular_column:
+            return data.other_columns[data.sel.index_of(*cdef)];
+        default:
+            throw exceptions::unsupported_operation_exception("Unknown column kind");
+        }
+    }
+}
+
+/// The comparator type for cv.
+const abstract_type* comparator(const column_value& cv) {
+    auto col_type = cv.col->type;
+    if (cv.sub) {
+        return static_pointer_cast<const collection_type_impl>(col_type)->value_comparator().get();
+    }
+    return col_type->is_reversed() ? col_type->underlying_type().get() : col_type.get();
+}
+
+/// Returns a tuple-valued terminal from t, if possible.  Otherwise, returns null.
+::shared_ptr<terminal> get_tuple(const ::shared_ptr<term> t, const query_options& opts) {
+    auto tml = dynamic_pointer_cast<terminal>(t);
+    if (tml) {
+        return tml;
+    }
+    auto marker = dynamic_pointer_cast<tuples::marker>(t);
+    if (marker) {
+        return marker->bind(opts);
+    }
+    auto delayed = dynamic_pointer_cast<tuples::delayed_value>(t);
+    if (delayed) {
+        return delayed->bind(opts);
+    }
+    return nullptr;
+}
+
+/// True iff lhs's value equals rhs.
+bool equal(const bytes_opt& rhs, const column_value& lhs, row_data data) {
+    if (!rhs) {
+        return false;
+    }
+    const auto value = get_value(lhs, data);
+    if (!value) {
+        return false;
+    }
+    return comparator(lhs)->equal(*value, *rhs);
+}
+
+/// True iff columns' values equal t.
+bool equal(::shared_ptr<term> t, const std::vector<column_value>& columns, row_data data) {
+    if (columns.size() > 1) {
+        auto multi = dynamic_pointer_cast<multi_item_terminal>(get_tuple(t, data.options));
+        if (!multi) {
+            throw exceptions::invalid_request_exception("RHS for multi-column = is not a tuple");
```

Comment:
nitpick: is it an error that users can actually see when putting something weird via CQL, or should it be a logic_error like the "empty tuple" error below? If the former, then I think it needs rephrasing - RHS is a TLA, so not everyone would understand what it means, and it also took me a couple of seconds to understand that `=` describes an operator type, not an equality sign meaning that `RHS for multi-column` is equivalent to `is not a tuple`, whatever that would mean :)

Reply:
I don't think this is reachable with the current validation upstream, but I like throwing invalid_request_exception here for staying future proof.  I'll reword the message to be useful to the end user, just in case they ever see it.

Reply:
Done.